src/integration/java/io/pinecone/integration/dataplane/UpdateAndQueryTest.java

```diff
@@ -0,0 +1,438 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static io.pinecone.helpers.BuildUpsertRequest.*;
+import static io.pinecone.helpers.IndexManager.createIndexIfNotExistsDataPlane;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class UpdateAndQueryTest {
+    private static VectorServiceGrpc.VectorServiceBlockingStub blockingStub;
+    private static VectorServiceGrpc.VectorServiceFutureStub futureStub;
+    private static final int dimension = 3;
+
+    @BeforeAll
+    public static void setUp() throws IOException, InterruptedException {
+        PineconeConnection connection = createIndexIfNotExistsDataPlane(dimension);
+        blockingStub = connection.getBlockingStub();
+        futureStub = connection.getFutureStub();
+    }
+
+    @Test
+    public void UpdateRequiredParamsFetchAndQuerySync() throws InterruptedException {
+        // Upsert vectors with required parameters
+        List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+        String namespace = RandomStringBuilder.build("ns", 8);
+        blockingStub.upsert(buildRequiredUpsertRequest(upsertIds, namespace));
+        Thread.sleep(3500);
+
+        // Verify the upserted vector count with fetch
+        FetchRequest fetchRequest = FetchRequest.newBuilder().addAllIds(upsertIds).setNamespace(namespace).build();
+        FetchResponse fetchResponse = blockingStub.fetch(fetchRequest);
+        assertEquals(fetchResponse.getVectorsCount(), upsertIds.size());
+        for (String key : upsertIds) {
+            assert (fetchResponse.containsVectors(key));
+        }
+
+        List<Float> updatedValues = Arrays.asList(101F, 102F, 103F);
+        // Update required fields only
+        UpdateRequest updateRequest = UpdateRequest.newBuilder()
+                .setId(upsertIds.get(0))
+                .setNamespace(namespace)
+                .addAllValues(updatedValues)
+                .build();
+
+        blockingStub.update(updateRequest);
+
+        // Query by ID to verify
+        QueryRequest queryRequest = QueryRequest.newBuilder()
+                .setId(upsertIds.get(0))
+                .setNamespace(namespace)
+                .setTopK(1)
+                .setIncludeValues(true)
+                .build();
+        QueryResponse queryResponse = blockingStub.query(queryRequest);
+        List<Float> queryResponseValues = queryResponse.getMatches(0).getValuesList();
+
+        assertEquals(updatedValues.size(), queryResponseValues.size());
+        int expectedValueSum = 0, actualValueSum = 0;
+        for (int i = 0; i < updatedValues.size(); i++) {
+            expectedValueSum += updatedValues.get(i);
+            actualValueSum += queryResponseValues.get(i);
+        }
+        assertEquals(expectedValueSum, actualValueSum);
```

Comment:
Summing seems fine, are we able to assert equality on the lists themselves without summing?

Reply:
Yes, this needs to go away, good catch. It was something I thought initially but I'm comparing the list directly.